### PR TITLE
Update PointAnnotation

### PR DESCRIPTION
**Pull request by frankrowe**
This PR is a proposal to enable PointAnnotations on Android by replacing deprecated classes with the [Mapbox Annotation Plugin](https://docs.mapbox.com/android/plugins/overview/annotation/).

Also added a new `draggable` prop to enable dragging interactions on PointAnnotations (for iOS and Android). Added `onDragStart` and `onDragEnd` events.

The following features were tested:
- [x] Default pin annotations Android
- [x] Custom view annotations Android
- [x] Draggable annotations Android
- [x] Callouts for Android
- [x] Default pin annotations iOS
- [x] Custom view annotations iOS
- [x] Draggable annotations iOS
- [x] Callouts for iOS

Implementation Details:
I followed this example: https://docs.mapbox.com/android/maps/examples/symbol-layer-info-window/ to get custom view annotations working. It converts the view to a bitmap and adds it to the Mapbox Style as an icon-image. I am not a Java expert so there may be a more performant way to do this.

### CODE REVIEW

File: ios/RCTMGL/RCTMGLPointAnnotation.m [hunk 1]
```diff
@@ -207,4 +214,65 @@ - (BOOL)_isFrameSet
...
+}
+
+- (void)startDragging {
+    [UIView animateWithDuration:0.3 delay:0 usingSpringWithDamping:0.5 initialSpringVelocity:0 options:0 animations:^{
+        self.layer.opacity = 0.8f;
+        self.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1.5, 1.5);
+    } completion:nil];
+    
+    // Initialize haptic feedback generator and give the user a light thud.
+    if (@available(iOS 10.0, *)) {
```
Comment (kristfal) [inline ios/RCTMGL/RCTMGLPointAnnotation.m]: I'd suggest we leave this out and let user's who want this experience on iOS to run it via a haptic plugin from the JS callbacks.

File: javascript/components/PointAnnotation.js [hunk 2]
```diff
@@ -43,6 +43,12 @@ class PointAnnotation extends React.PureComponent {
      */
     selected: PropTypes.bool,
 
+    /**
+     * Enable or disable dragging. Defaults to false.
+     * @type {[type]}
```
Comment (kristfal) [inline javascript/components/PointAnnotation.js]: This comment can be removed as it passes down to the docs. The actual type is parsed from PropType.

File: ios/RCTMGL/RCTMGLPointAnnotation.m (hunk 1 above)
Comment (frankrowe) [reply ios/RCTMGL/RCTMGLPointAnnotation.m]: Removed. I also reduce opacity and increase size of the view in this func. Should we remove that also?

Comment (kristfal) [reply ios/RCTMGL/RCTMGLPointAnnotation.m]: Yes, I think we should do that or expose some props to toggle this behaviour. 

I believe the most important part is to make iOS and Android behave as close to 1:1 as possible.

**Comment by kristfal** [comment]
Thanks a lot @frankrowe and super work on this. Would like to get @mfazekas's 👀 and an iOS/Android feature alignment on this before we merge.

File: ios/RCTMGL/RCTMGLPointAnnotation.m (hunk 1 above)
Comment (frankrowe) [reply ios/RCTMGL/RCTMGLPointAnnotation.m]: Removed all dragging effects.

**Comment by frankrowe** [comment]
@kristfal @mfazekas any other work you think needs to be done to get this merged?

**Comment by kristfal** [comment]
Imo this is good to merge. We could follow up with a dragable marker example, but that is bonus imo :) merging.

**Pull request merged**